tests/testUtil.py

```diff
@@ -152,7 +151,37 @@ def registerDatasetTypes(registry, pipeline):
                 registry.registerDatasetType(datasetType)
 
 
-def makeSimpleQGraph(nQuanta=5, pipeline=None, butler=None, root=None, skipExisting=False):
+def _makeButlerRepo(root, inMemory=True):
```

Comment:
Why did you stop using `lsst.daf.butler.tests.makeTestRepo` -- it did this configuration for you.

Reply:
makeTestRepo only makes in-memory repo, for this new feature I have to test with persistent repo because I need to instantiate multiple butlers/registries on the same data.

Reply:
You can pass a Config of overrides into it. For example as done in https://github.com/lsst/obs_base/blob/master/tests/test_butlerFits.py#L111  -- you can set just the registry.db part of a Config.

Reply:
Well, I'll have to pass something for `datastore.cls` which I don't know what it has to be. Or will it work if I set it to None? Besides what I really need is a butler config, in I was thinking that `butler._config` is not public and it may be better not to use it (and I did not check that it's unmodified from what is passed originally).

Reply:
Do you really need butler_config? The use cases below would work if you just passed self.root wouldn't they? (since the butler would read the on-disk butler.yaml file from self.root. I'm probably missing something. There will always be a butler.yaml written into root given the options specified there (other options allow it to be written outside). I'm not going to insist on shoehorning this into makeButlerRepo though if there is a good reason not to do it.

Reply:
OK, I re-implemented that part and also my new test code, it does not need to instantiate multiple butlers now (but needs to call some internal methods to refresh collections). I hope tests are getting better, thanks for suggestions!